### ISSUE TITLE
Analysis report is not getting updated in SonarQube

### ISSUE THREAD

**Issue by AswinVeer** [gretard/sonar-ps-plugin#28]
Hello!

Happy New Year :)

We have installed the sonar-ps-plugin in our SonarQube server and tried running an analysis. The plugin detects the ps files and lines of code in it, but no analysis report like code smells, bugs etc. Then we realized that PSScriptAnalyzer is missing from the below steps and prompt note (as attached), but not sure how to make it work so that SonarQube can throw the analysis report as we expect.

- Download and install SonarQube
- Download plugin from the [releases](https://github.com/gretard/sonar-ps-plugin/releases) and copy it to sonarqube's extensions\plugins directory
- Start SonarQube and enable rules
- Install [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) into your build machine where you plan to run sonar scanner

Please advise! Thanks in Advance!

![image](https://user-images.githubusercontent.com/80384953/210411897-760178c0-9050-4845-882a-01b1ac696f6f.png)

**Comment by AswinVeer** [comment]
Just add-on to my previous note.. 

Specifically, I need to know how to make analyzer work in build machine and CI/CD pipeline.

Thanks!

**Comment by AswinVeer** [comment]
Hi @gretard , @moerketh , @caminsha 

Could one of you please answer my question?

Thanks!

**Comment by moerketh** [comment]
From your screenshot it looks like PSScriptAnalyzer could not be found. Does it help to run `Install-Module -Name PSScriptAnalyzer -Force` on that machine? What PowerShell version are you running (check with `$PSVersionTable` )?

**Comment by AswinVeer** [comment]
@moerketh  Thanks for replying! 

Do you have any specific documentation that can help me on this? I am not sure what I am missing. 

Also, I do have two queries - 

Firstly, Is this something can be able to perform only during local run? I mean, running a local SonarQube analysis with analyzer installed on the build machine.

Secondly, how it will work in CI/CD pipeline, I mean in Jenkins or Tekton. Please advise!

**Comment by AswinVeer** [comment]
Hi @moerketh  Any update on my previous comment pls?

**Comment by gretard** [comment]
Hi @AswinVeer ,

Let's see if you have everything configured:
1. Did you install PSScriptAnalyzer onto your build machine? Instructions are here https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/overview?view=ps-modules#installing-psscriptanalyzer, for example open powershell terminal and run:

```
Install-Module -Name PSScriptAnalyzer -Force
``` 

2. Check if PSScriptANalyzer is working onto your build machine (https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/using-scriptanalyzer?source=recommendations&view=ps-modules). After running 1 step again in powershell terminal run: 

```
Invoke-ScriptAnalyzer -ScriptDefinition '"b" = "b"; function eliminate-file () { }'
```
Please let me know if after having performed first step second is working. If yes then plugin should be working fine. If not - please provide debug logs by running sonar scanner using -X flag to get the debug output.

**Comment by AswinVeer** [comment]
Hi @gretard,

Thanks much!

Your steps helped me to figure out the issue and sonar analysis went fine now. 

Following to this, do you have any suggestion for us on importing the code coverage results? Please advise!

**Comment by gretard** [comment]
@AswinVeer  - this is open source plugin and it does not support coverage reports now. 

If you need commercial support - please let me know and we can arrange something. Otherwise - if there is a need for coverage reports you can try using sonar scanner capabilities to import coverage reports in generic format. You can read more at  https://docs3.sonarqube.org/latest/analysis/coverage/ and https://docs3.sonarqube.org/latest/analysis/generic-test/

**Comment by AswinVeer** [comment]
Hi @gretard - Thanks for your reply!

Would you be able to tell us how the setup works in pipeline? I mean Jenkins/Tekton.. Please advise!
